src/lib.rs

```diff
@@ -76,6 +83,54 @@ impl ErrorCode for m::rstatus_et {
     }
 }
 
+#[derive(Debug)]
```

Comment:
```suggestion
#[derive(Debug)]
#[non_exhaustive]
```
It's not unlikely that there are other hidden invariants, so it will be helpful if adding additional error kinds is not a breaking change.

Reply:
Oh I didn't actually catch that the enum isn't exposed. I do wonder if it should be, however. It is certainly more in line with the original library to give very little feedback that the program can do something with, but that's partially a limitation of C. 

I've personally used METIS to partition user-supplied `.obj` files, and only some of the errors that could come up would be a programming error rather than malformed user input. In that case, the user has no idea what `xadj` or `adjncy` means.

I don't actually have any opinion about which way is better, I just don't see any particular benefit to keeping it private.

Reply:
With <https://github.com/LIHPC-Computational-Geometry/metis-rs/pull/13/commits/41a1d76d53c9ac56aa58024edeaa2f8927c8d501> the CSR terms in error messages now match `Graph::new`'s documentation.

I kept the errors related to the CSR format opaque. If needed it should be possible to make a more detailed API later without breaking changes, but I don't see how these can be acted upon at runtime.